include/mtx/filters.hpp

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#if __has_include(<nlohmann/json_fwd.hpp>)
+#include <nlohmann/json_fwd.hpp>
+#else
+#include <nlohmann/json.hpp>
+#endif
+
+#include <string>
+
+//! Filters can be created on the server and can be
+//! passed as a parameter to APIs which return events.
+
+namespace mtx {
+namespace filters {
```

Comment:
I'd use public_rooms as the namespace or since it is that small, just put it in the requests header and call it public_rooms::Filter or PublicRoomsFilter.

Reply:
sure, the main reason i created the namespace is if we decide to extend and use other filters (like `RoomFilter`, `StateFilter`) in the future.

Reply:
Well, the problem is that this filter is not reusable outside of the room directory requests, so it makes no sense to put it somewhere to reuse. We can move it, if that becomes the case.